### PR TITLE
Change format of corpus snapshots to save cost

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Don't store every file on each snapshot, just newly modified ones.
Also, get rid of unchanged-cycles, it's no longer as useful and adds complexity.

### CODE REVIEW

File: experiment/measurer/test_run_coverage.py [hunk 1]
```diff
@@ -66,12 +66,13 @@ def test_integration_do_coverage_run_crash(self, tmp_path):
...
-        crashing_units = run_coverage.do_coverage_run(self.COVERAGE_BINARY_PATH,
-                                                      units, profraw_file,
-                                                      crashes_dir)
+        run_coverage.do_coverage_run(self.COVERAGE_BINARY_PATH, units,
+                                     profraw_file, crashes_dir)
 
         # Ensure the crashing units are returned.
-        assert crashing_units == ['86f7e437faa5a7fce15d1ddcb9eaeaea377667b8']
+        assert os.listdir(crashes_dir) == [
+            '86f7e437faa5a7fce15d1ddcb9eaeaea377667b8'
```
Comment (Alan32Liu) [inline experiment/measurer/test_run_coverage.py:74]: CI failure suggest this name should be prefixed by 'crash-'.

**Comment by Alan32Liu** [comment]
Shall we merge this after the competition evaluation? Or test it in a separate branch for a while before merging it into ` `master`?

**Comment by Alan32Liu** [comment]
> > Shall we merge this after the competition evaluation? Or test it in a separate branch for a while before merging it into ` `master`?
> 
> I wonder if we should create a new "stable" branch and set that as the default branch and have all ongoing experiments/competition use that.

Yes! That would be ideal.
Yesterday I noticed some online experiments do not seem to start properly again, I am planning to look into them later.

**Comment by jonathanmetzman** [comment]
> Shall we merge this after the competition evaluation? Or test it in a separate branch for a while before merging it into `master`?

I think this should come before the competition. And ASAP frankly. This is money down the 🚽

**Comment by jonathanmetzman** [comment]
> > Shall we merge this after the competition evaluation? Or test it in a separate branch for a while before merging it into ` `master`?
> 
> I wonder if we should create a new "stable" branch and set that as the default branch and have all ongoing experiments/competition use that.

I think https://github.com/google/fuzzbench/issues/1599 can enable this.

**Comment by jonathanmetzman** [comment]
Ping

**Comment by Alan32Liu** [comment]
> #1599

If we want to merge it before the competition, I am happy to merge it ASAP : )

**Comment by Alan32Liu** [comment]
> Ping

If we want to merge it before the competition, I am happy to merge it ASAP : )

**Comment by jonathanmetzman** [comment]
> > Ping
> 
> If we want to merge it before the competition, I am happy to merge it ASAP : )
> 
> Shall we run some test experiments to verify if this works as expected?

Running test experiments is a good idea.
I definitely want this merged before the competition. Otherwise we will pay a lot more for the competition.

**Comment by Alan32Liu** [comment]
> > > Ping
> > 
> > 
> > If we want to merge it before the competition, I am happy to merge it ASAP : )
> > Shall we run some test experiments to verify if this works as expected?
> 
> Running test experiments is a good idea. I definitely want this merged before the competition. Otherwise we will pay a lot more for the competition.

Great!
I will launch one test experiment locally today.

**Comment by jonathanmetzman** [comment]
Thank you thank you this is really above and beyond

**Comment by jonathanmetzman** [comment]
> Thank you thank you this is really above and beyond

What were the results of this experiment?

**Comment by Alan32Liu** [comment]
> > Thank you thank you this is really above and beyond
> 
> What were the results of this experiment?

## Experiment 1 [`2023-01-27-test-request`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-request;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
Launched from `experiment-request.yaml`
The experiment is **_still running_**.

## Experiment 2 [`2023-01-27-test-local-launch`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-local-launch;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
Launched from the local command line.
There are **_many build failures_**, as described in #1648.

## Experiment 3 [`2023-01-30-test-local-launch-1`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-30-test-local-launch-1;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
Launch from the local command line with higher memory worker pool instances, following the discussion in #1648.
**_Build failure disappeared_**.

**Comment by jonathanmetzman** [comment]
> > > Thank you thank you this is really above and beyond
> > 
> > 
> > What were the results of this experiment?
> 
> ## Experiment 1 [`2023-01-27-test-request`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-request;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> Launched from `experiment-request.yaml` The experiment is **_still running_**.
> 
> ## Experiment 2 [`2023-01-27-test-local-launch`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-local-launch;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> Launched from the local command line. There are **_many build failures_**, as described in #1648.
> 
> ## Experiment 3 [`2023-01-30-test-local-launch-1`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-30-test-local-launch-1;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> Launch from the local command line with higher memory worker pool instances, following the discussion in #1648. **_Build failure disappeared_**.

So all of them failed. Can you do new experiments so we can test this please?

**Comment by Alan32Liu** [comment]
> 2023-01-30-test-local-launch-1



> > > > Thank you thank you this is really above and beyond
> > > 
> > > 
> > > What were the results of this experiment?
> > 
> > 
> > ## Experiment 1 [`2023-01-27-test-request`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-request;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > Launched from `experiment-request.yaml` The experiment is **_still running_**.
> > ## Experiment 2 [`2023-01-27-test-local-launch`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-local-launch;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > Launched from the local command line. There are **_many build failures_**, as described in #1648.
> > ## Experiment 3 [`2023-01-30-test-local-launch-1`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-30-test-local-launch-1;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > Launch from the local command line with higher memory worker pool instances, following the discussion in #1648. **_Build failure disappeared_**.
> 
> So all of them failed. Can you do new experiments so we can test this please?

No all of them, I suppose: The last one works fine after using a high-memory worker instance.
I can run a 24-hour experiment with all core fuzzers to double-check everything if that sounds good to you?

**Comment by jonathanmetzman** [comment]
> > 2023-01-30-test-local-launch-1
> 
> > > > > Thank you thank you this is really above and beyond
> > > > 
> > > > 
> > > > What were the results of this experiment?
> > > 
> > > 
> > > ## Experiment 1 [`2023-01-27-test-request`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-request;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > > Launched from `experiment-request.yaml` The experiment is **_still running_**.
> > > ## Experiment 2 [`2023-01-27-test-local-launch`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-27-test-local-launch;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > > Launched from the local command line. There are **_many build failures_**, as described in #1648.
> > > ## Experiment 3 [`2023-01-30-test-local-launch-1`](https://pantheon.corp.google.com/storage/browser/www.fuzzbench.com/reports/2023-01-30-test-local-launch-1;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false)
> > > Launch from the local command line with higher memory worker pool instances, following the discussion in #1648. **_Build failure disappeared_**.
> > 
> > 
> > So all of them failed. Can you do new experiments so we can test this please?
> 
> No all of them, I suppose: The last one works fine after using a high-memory worker instance. I can run a 24-hour experiment with all core fuzzers to double-check everything if that sounds good to you?

Run it with a few fuzzers, we don't need all of them.

**Comment by jonathanmetzman** [comment]
Damn, where did these test failures come from.

**Comment by jonathanmetzman** [comment]
> Damn, where did these test failures come from.

Think its related to that start cycle at 0. If it breaks this and it's too difficult to fix I might revert it. This change is super important.

**Comment by jonathanmetzman** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name test-storage-change --fuzzers afl libfuzzer centipede honggfuzz

**Comment by jonathanmetzman** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name test-storage-change --fuzzers afl libfuzzer centipede honggfuzz

**Comment by jonathanmetzman** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name test-storage-change --fuzzers afl libfuzzer centipede honggfuzz -a

**Comment by jonathanmetzman** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name test-storage-change2 --fuzzers afl aflplusplus libfuzzer centipede honggfuzz -a

**Comment by jonathanmetzman** [comment]
test-storage-change is still running.

**Comment by Alan32Liu** [comment]
@jonathanmetzman `make format` failed : )

**Pull request merged**